app/scripts/controllers/preferences-controller.ts

```diff
@@ -17,32 +33,150 @@ const testNetworks = {
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions =
+  | AccountsControllerGetAccountByAddressAction
+  | AccountsControllerSetAccountNameAction
+  | AccountsControllerGetSelectedAccountAction
+  | AccountsControllerSetSelectedAccountAction;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents = AccountsControllerChangeEvent;
+export type PreferencesControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  PreferencesControllerActions | AllowedActions,
+  PreferencesControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
+type PreferencesControllerOptions = {
+  networkConfigurations?: Record<string, { chainId: Hex }>;
+  initState?: Partial<PreferencesControllerState>;
+  initLangCode?: string;
+  messenger: PreferencesControllerMessenger;
+};
+
+type Preferences = {
+  autoLockTimeLimit?: number;
+  showExtensionInFullSizeView: boolean;
+  showFiatInTestnets: boolean;
+  showTestNetworks: boolean;
+  smartTransactionsOptInStatus: boolean | null;
+  useNativeCurrencyAsPrimaryCurrency: boolean;
+  hideZeroBalanceTokens: boolean;
+  petnamesEnabled: boolean;
+  redesignedConfirmationsEnabled: boolean;
+  redesignedTransactionsEnabled: boolean;
+  featureNotificationsEnabled: boolean;
+  isRedesignedConfirmationsDeveloperEnabled: boolean;
+  showConfirmationAdvancedDetails: boolean;
+};
+
+export type PreferencesControllerState = {
+  selectedAddress: string;
+  useBlockie: boolean;
+  useNonceField: boolean;
+  usePhishDetect: boolean;
+  dismissSeedBackUpReminder: boolean;
+  useMultiAccountBalanceChecker: boolean;
+  useSafeChainsListValidation: boolean;
+  useTokenDetection: boolean;
+  useNftDetection: boolean;
+  use4ByteResolution: boolean;
+  useCurrencyRateCheck: boolean;
+  useRequestQueue: boolean;
+  openSeaEnabled: boolean;
+  securityAlertsEnabled: boolean;
+  watchEthereumAccountEnabled: boolean;
+  bitcoinSupportEnabled: boolean;
+  bitcoinTestnetSupportEnabled: boolean;
+  addSnapAccountEnabled: boolean;
+  advancedGasFee: Record<string, Record<string, string>>;
+  featureFlags: Record<string, boolean>;
+  incomingTransactionsPreferences: Record<number, boolean>;
+  knownMethodData: Record<string, string>;
+  currentLocale: string;
+  identities: Record<string, AccountIdentityEntry>;
+  lostIdentities: Record<string, object>;
+  forgottenPassword: boolean;
+  preferences: Preferences;
+  ipfsGateway: string;
+  isIpfsGatewayEnabled: boolean;
+  useAddressBarEnsResolution: boolean;
+  ledgerTransportType: LedgerTransportTypes;
+  snapRegistryList: Record<string, object>;
+  theme: ThemeType;
+  snapsAddSnapAccountModalDismissed: boolean;
+  useExternalNameSources: boolean;
+  useTransactionSimulations: boolean;
+  enableMV3TimestampSave: boolean;
+  useExternalServices: boolean;
+  textDirection?: string;
+};
+
 export default class PreferencesController {
```

Comment:
nit: What do you think about aligning this export statement with the core controller here? The extra diffs for going back and forth between default vs. named import seem unnecessary.

```suggestion
export class PreferencesController {
```

```diff
diff --git a/app/scripts/controllers/mmi-controller.ts b/app/scripts/controllers/mmi-controller.ts
index 7cc024d0cd..4b1e534907 100644
--- a/app/scripts/controllers/mmi-controller.ts
+++ b/app/scripts/controllers/mmi-controller.ts
@@ -41,7 +41,7 @@ import AccountTracker from '../lib/account-tracker';
 import { getCurrentChainId } from '../../../ui/selectors';
 import MetaMetricsController from './metametrics';
 import { getPermissionBackgroundApiMethods } from './permissions';
-import PreferencesController from './preferences-controller';
+import { PreferencesController } from './preferences-controller';
 import { AppStateController } from './app-state';
 
 type UpdateCustodianTransactionsParameters = {
diff --git a/app/scripts/controllers/preferences-controller.test.ts b/app/scripts/controllers/preferences-controller.test.ts
index 56885232be..1822071e22 100644
--- a/app/scripts/controllers/preferences-controller.test.ts
+++ b/app/scripts/controllers/preferences-controller.test.ts
@@ -14,12 +14,13 @@ import { SnapControllerStateChangeEvent } from '@metamask/snaps-controllers';
 import { CHAIN_IDS } from '../../../shared/constants/network';
 import { mockNetworkState } from '../../../test/stub/networks';
 import { ThemeType } from '../../../shared/constants/preferences';
-import PreferencesController, {
+import type {
   AllowedActions,
   AllowedEvents,
   PreferencesControllerActions,
   PreferencesControllerEvents,
 } from './preferences-controller';
+import { PreferencesController } from './preferences-controller';
 
 const NETWORK_CONFIGURATION_DATA = mockNetworkState(
   {
diff --git a/app/scripts/lib/ppom/ppom-middleware.ts b/app/scripts/lib/ppom/ppom-middleware.ts
index d623d4868a..d3cbea90f3 100644
--- a/app/scripts/lib/ppom/ppom-middleware.ts
+++ b/app/scripts/lib/ppom/ppom-middleware.ts
@@ -11,7 +11,7 @@ import { detectSIWE } from '@metamask/controller-utils';
 
 import { MESSAGE_TYPE } from '../../../../shared/constants/app';
 import { SIGNING_METHODS } from '../../../../shared/constants/transaction';
-import PreferencesController from '../../controllers/preferences-controller';
+import { PreferencesController } from '../../controllers/preferences-controller';
 import { AppStateController } from '../../controllers/app-state';
 import { LOADING_SECURITY_ALERT_RESPONSE } from '../../../../shared/constants/security-provider';
 import { getProviderConfig } from '../../../../ui/ducks/metamask/metamask';
diff --git a/app/scripts/metamask-controller.js b/app/scripts/metamask-controller.js
index b7214ad198..478e0cc750 100644
--- a/app/scripts/metamask-controller.js
+++ b/app/scripts/metamask-controller.js
@@ -283,7 +283,7 @@ import { NetworkOrderController } from './controllers/network-order';
 import { AccountOrderController } from './controllers/account-order';
 import createOnboardingMiddleware from './lib/createOnboardingMiddleware';
 import { isStreamWritable, setupMultiplex } from './lib/stream-utils';
-import PreferencesController from './controllers/preferences-controller';
+import { PreferencesController } from './controllers/preferences-controller';
 import AppStateController from './controllers/app-state';
 import AlertController from './controllers/alert';
 import OnboardingController from './controllers/onboarding';
diff --git a/shared/constants/mmi-controller.ts b/shared/constants/mmi-controller.ts
index 67123b83d5..0efcceeb08 100644
--- a/shared/constants/mmi-controller.ts
+++ b/shared/constants/mmi-controller.ts
@@ -4,7 +4,7 @@ import { TransactionUpdateController } from '@metamask-institutional/transaction
 import { CustodyController } from '@metamask-institutional/custody-controller';
 import { SignatureController } from '@metamask/signature-controller';
 import { NetworkController } from '@metamask/network-controller';
-import PreferencesController from '../../app/scripts/controllers/preferences-controller';
+import { PreferencesController } from '../../app/scripts/controllers/preferences-controller';
 import { AppStateController } from '../../app/scripts/controllers/app-state';
 import AccountTracker from '../../app/scripts/lib/account-tracker';
 import MetaMetricsController from '../../app/scripts/controllers/metametrics';
```